npm/pkg/dataplane/ipsets/ipset.go

```diff
@@ -0,0 +1,191 @@
+package ipsets
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+type IPSet struct {
+	Name       string
+	HashedName string
+	Type       SetType
+	// IpPodKey is used for setMaps to store Ips and ports as keys
+	// and podKey as value
+	IPPodKey map[string]string
+	// This is used for listMaps to store child IP Sets
+	MemberIPSets map[string]*IPSet
+	// Using a map here to emulate set of netpol names
+	SelectorReference map[string]struct{}
+	NetPolReference   map[string]struct{}
+	IpsetReferCount   int
+}
+
+type SetType int32
+
+const (
+	Unknown                  SetType = 0
+	NameSpace                SetType = 1
+	KeyLabelOfNameSpace      SetType = 2
+	KeyValueLabelOfNameSpace SetType = 3
+	KeyLabelOfPod            SetType = 4
+	KeyValueLabelOfPod       SetType = 5
+	NamedPorts               SetType = 6
+	NestedLabelOfPod         SetType = 7
+	CIDRBlocks               SetType = 8
+)
+
+var SetTypeName = map[int32]string{
+	0: "Unknown",
+	1: "NameSpace",
+	2: "KeyLabelOfNameSpace",
+	3: "KeyValueLabelOfNameSpace",
+	4: "KeyLabelOfPod",
+	5: "KeyValueLabelOfPod",
+	6: "NamedPorts",
+	7: "NestedLabelOfPod",
+	8: "CIDRBlocks",
+}
+
+var SetTypeValue = map[string]int32{
+	"Unknown":                  0,
+	"NameSpace":                1,
+	"KeyLabelOfNameSpace":      2,
+	"KeyValueLabelOfNameSpace": 3,
+	"KeyLabelOfPod":            4,
+	"KeyValueLabelOfPod":       5,
+	"NamedPorts":               6,
+	"NestedLabelOfPod":         7,
+	"CIDRBlocks":               8,
+}
+
+func (x SetType) String() string {
+	return SetTypeName[int32(x)]
+}
+
+func GetSetType(x string) SetType {
+	return SetType(SetTypeValue[x])
+}
+
+type SetKind string
+
+const (
+	ListSet SetKind = "list"
+	HashSet SetKind = "set"
```

Comment:
tbh I think there's room for what defines a `set` here, taking linux vernacular could be misleading in windows, but I think in terms of platform agnostic, what about something like hashset is "set" and list is "superset" since it contains only "sets"(?)

Reply:
I like this "superset" since it is intuitive. But keeping list since we have been used to calling it as list. We can use "nestedset" which windows is using ?